pandasdmx/reader/sdmxml.py

```diff
@@ -466,6 +466,9 @@ def _maintained(self, cls=None, id=None, urn=None, **kwargs):
             cls = get_class(match['package'], match['class'])
             id = match['id']
 
+            # Re-add the URN to the kwargs
+            kwargs['urn'] = urn
+
         key = (cls.__name__, id) if isclass(cls) else (cls, id)
 

```

Comment:
This comments is about line 484: 
```python
assert kwargs.pop('is_external_reference', True)
```

You know by now my pop-aversion. The check is wierd. It raises and error if and only if 'is_external_reference' is in kwargs and evaluates to False. If it is not in kwargs, the assertion passes. The correct assert condition, in my view is:

`assert kwargs['is_external_reference']`

This checks for the existence of the key incidentally, and whether it is True as it should be. If you are concerned with a KeyError, use kwarts.get() instead. Upon instantiation in line 485, ìs_external_reference` is in the kwargs so you don't have to pass it explicitely.

Reply:
> It raises [an] error if and only if 'is_external_reference' is in kwargs and evaluates to False.

Correct! If the code gets to this point under those conditions, something very weird *is* going on with the message being parsed, and we can't continue.

This XML attribute is more frequently absent then present, so we can't simply lookup. But you remind me that dicts also have [setdefault()](https://docs.python.org/3/library/stdtypes.html#dict.setdefault), which we can use instead while also shortening the code.